src/main/ruby/truffleruby/core/truffle/feature_loader.rb

```diff
@@ -140,6 +143,16 @@ def self.find_feature_or_file(feature, use_feature_provided = true)
       end
     end
 
+    def self.load_unless_realpath_loaded(feature, path)
+      # TODO: does this need to be synchronized?
+      realpath = (@features_realpath_cache[path] ||= File.realpath(path) || path)
```

Comment:
AFAIK File.realpath never returns false|nil, always a String, or an exception if some part does not exist.

Reply:
That did seem to be the case for me but I wasn't sure if it was true on all platforms.
I think I was looking at this C code and it made me want to program defensively (the semantics are probably different, though).
https://github.com/ruby/ruby/blob/7ce4b716bdb5bcfc8b30ffcd034ce7aded1f72b9/load.c#L87-L88